src/storage-client/src/sink.rs

```diff
@@ -420,17 +420,24 @@ pub async fn determine_latest_progress_record(
         C: ConsumerContext,
     {
         // ensure the progress topic has exactly one partition
-        let partitions = mz_kafka_util::client::get_partitions(
+        let partitions = match mz_kafka_util::client::get_partitions(
             progress_client.client(),
             progress_topic,
             timeout,
-        )
-        .with_context(|| {
-            format!(
-                "Unable to fetch metadata about progress topic {}",
-                progress_topic
-            )
-        })?;
+        ) {
+            Ok(partitions) => partitions,
+            Err(GetPartitionsError::TopicDoesNotExist) => {
+                // The progress topic doesn't exist, which indicates there is
+                // no committed timestamp.
+                return Ok(None);
```

Comment:
This might not be quite right because when we determine the latest progress record, we are "sure" that we have already created the progress topic. If the topic DNE, there must have been a delay in propagating the topic's creation to the broker. I think we should probably await the topic to come into existence rather than assume the progress record DNE.

However, this might be wrong in light of my other questions.

Reply:
Yeah, see my top-level response! AFAICT, there's nothing that guarantees that the progress topic exists when we call `determine_latest_progress_record`.

Reply:
I think we're on the same page here. I was referring to the fact that the progress topic _should eventually_ exist at this point (sorry for the imprecision). However, that shouldn't actually matter because the non-existence of the progress topic suggests that no values could have been written to it.